kuma/javascript/src/payments/feedback-form.jsx

```diff
@@ -0,0 +1,56 @@
+// @flow
+import * as React from 'react';
+import { gettext } from '../l10n.js';
+import GAProvider from '../ga-provider.jsx';
+
+const FeedbackForm = () => {
+    const [feedback, setFeedback] = React.useState('');
+    const [showMessage, setShowMessage] = React.useState(false);
+    const ga = React.useContext(GAProvider.context);
+
+    const handleChange = event => {
+        const { value } = event.target;
+        setFeedback(value);
+
+        // cases where user submitted feedback already
+        // and start typing in text input again
+        if (showMessage) {
+            setShowMessage(false);
+        }
+    };
+
+    const handleSubmit = event => {
+        event.preventDefault();
+        ga('send', {
+            hitType: 'event',
+            eventCategory: 'monthly payments',
+            eventAction: 'feedback',
+            eventLabel: feedback.trim()
+        });
+
+        // Clear form and show thank you message
+        setFeedback('');
+        setShowMessage(true);
+    };
+
+    return (
+        <form onSubmit={handleSubmit}>
+            <input
+                type="text"
+                placeholder={gettext('Enter optional feedback…')}
+                name="feedback"
+                value={feedback}
+                onChange={handleChange}
+                required
```

Comment:
Not very optional then, is it? :)

Reply:
I notice that you can submit and empty string as valid feedback. Or essentially anything as long as there is at least one character.

Reply:
Good catch-- are there guidelines around what we should accept, e.g. string length?

Reply:
I think a good rule of them is that it needs to be something more than `feedback.trim()`. So an empty string or a bunch of whitespace should NOT submit.

Reply:
Agreed not being able to submit empty feedback is a good start. The next level up is always tricky. We do want useful feedback I reckon so maybe a 10 character minimum? Not sure who should make that call though :( I would say trust your gut on this one

Reply:
Thank you both-- I added a rule of minimum 10 characters.